client/dashboard/sites/settings-site-visibility/privacy-form.tsx

```diff
@@ -3,57 +3,115 @@ import {
 	__experimentalHStack as HStack,
 	__experimentalVStack as VStack,
 	Button,
+	CheckboxControl,
+	ExternalLink,
 } from '@wordpress/components';
-import { __ } from '@wordpress/i18n';
-import { useState } from 'react';
-import type { SiteSettings } from '../../data/types';
-import type { Field, SimpleFormField } from '@automattic/dataviews';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { useState, useMemo } from 'react';
+import type { SiteSettings, Site } from '../../data/types';
+import type { Field, Form, FormField } from '@automattic/dataviews';
 import type { UseMutationResult } from '@tanstack/react-query';
 
-const fields: Field< SiteSettings >[] = [
-	{
-		id: 'wpcom_site_visibility',
-		Edit: 'toggleGroup',
-		elements: [
-			{
-				label: __( 'Coming soon' ),
-				value: 'coming-soon',
-				description: __(
-					'Your site is hidden from visitors behind a "Coming Soon" notice until it is ready for viewing.'
-				),
-			},
-			{
-				label: __( 'Public' ),
-				value: 'public',
-				description: __( 'Your site is visible to everyone.' ),
-			},
-			{
-				label: __( 'Private' ),
-				value: 'private',
-				description: __(
-					'Your site is only visible to you and logged-in members you approve. Everyone else will see a log in screen.'
-				),
-			},
-		],
-	},
-];
-
-const form = {
-	type: 'regular' as const,
-	fields: [ { id: 'wpcom_site_visibility', labelPosition: 'none' } as SimpleFormField ],
-};
-
 export function PrivacyForm( {
+	site,
 	settings,
 	mutation,
 }: {
+	site: Site;
 	settings: SiteSettings;
 	mutation: UseMutationResult< Partial< SiteSettings >, Error, Partial< SiteSettings >, unknown >;
 } ) {
 	const [ formData, setFormData ] = useState( {
 		wpcom_site_visibility: settings.wpcom_site_visibility,
+		wpcom_discourage_search_engines: settings.wpcom_discourage_search_engines,
+		wpcom_prevent_third_party_sharing:
+			settings.wpcom_discourage_search_engines || settings.wpcom_prevent_third_party_sharing,
 	} );
 
+	const fields: Field< SiteSettings >[] = useMemo(
```

Comment:
`fields` had to become a part of the component because the site's domain is part of one of the labels, so `fields` needs access to the `site` prop.

Another way of doing this would be to add the site domain as a special secret property of the `DataForm`'s `Item` i.e. use a field type like:

```ts
Field< SiteSettings & { domain: string } >
```

This would allow me to pass the domain as a prop to the `Edit` component. But it might also feel like a hack. Memo also does the trick of preventing the `DataForm` from re-rendering every time a setting is touched.

Reply:
I started a discussion p1748483462190909-slack-C08JZTRS6NA

I don't see any other dynamic labels like this in the other settings mockups. It looks like this dynamic label was copied directly from Calypso, so there wasn't necessarily deep thought behind this.

I'm going to deviate from the mockup and use a static checkbox label. Depending on the outcome of the discussion then I can follow up with another PR.